### ISSUE TITLE
CA 1508 false positive when you check for null and then check for an empty string through the string.IsNullOrWhiteSpace method 

### ISSUE THREAD

**Issue by DmitryKondratenko1986** [dotnet/roslyn-analyzers#5305]
### Analyzer

**Diagnostic ID**: [CA1508](https://docs.microsoft.com/ru-ru/dotnet/fundamentals/code-analysis/quality-rules/ca1508): `Avoid dead conditional code`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 5.0.302](https://dotnet.microsoft.com/download/dotnet/5.0)

_AND_

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://github.com/DotNetAnalyzers/StyleCopAnalyzers)

**Version**: 1.1.118 (Latest)

### Describe the bug
In the code example below, the analyzer assumes that the second if-condition int the set-sections in Model and Owner properties can never be met, considering this code to be "dead"

### Steps To Reproduce

Provide the steps to reproduce the behavior:
1. Create class library project with .net 5.0
2. Add project property AnalysisMode: AllEnabledByDefault
3. Create a simple class (see the code example below)
4. See CA1508 warning in the second if-branch in the Owner and Model properties

### Sample
``` c#
public class Vehicle
{
    private string model;
    private string owner;

    public Vehicle(string model, string owner)
    {
        this.Model = model;
        this.Owner = owner;
    }

    public string Model
    {
        get => this.model;
        private set
        {
            if (value is null)
            {
                throw new ArgumentNullException(nameof(value));
            }

            if (string.IsNullOrWhiteSpace(value))
            {
                throw new ArgumentException($"Model cannot be empty or whitespace", nameof(value));
            }

            this.model = value;
        }
    }

    public string Owner
    {
        get => this.owner;
        set
        {
            if (value is null)
            {
                throw new ArgumentNullException(nameof(value));
            }

            if (string.IsNullOrWhiteSpace(value))
            {
                throw new ArgumentException($"Owner cannot be empty or whitespace", nameof(value));
            }

            this.owner = value;
        }
    }
}

```

### Expected behavior
No CA1508.

### Actual behavior
CA1508 false positive.

## Additional context
![изображение](https://user-images.githubusercontent.com/68240319/127188388-bdd78fe7-232f-4da8-8ab5-e94c4a063533.png)

I went into this branch of code using a debugger 
![изображение](https://user-images.githubusercontent.com/68240319/127188468-e1104dfc-813a-4798-981e-5ddef989ae98.png)

**Comment by mavasani** [comment]
This is likely fixed in the latest NuGet package/.NET 6 SDK

**Comment by mavasani** [comment]
Yep, confirmed this is fixed in .NET6 SDK
